### PR TITLE
T-25 smartrifle

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
T-25 smartrifle

## Why It's Good For The Game
sg variety

## Changelog
:cl:
expansion: Added new T-25 SmartRifle for smartgunners
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 1]
```diff
@@ -1081,7 +1081,9 @@ GLOBAL_LIST_INIT(marine_selector_cats, list(
 	req_access = list(ACCESS_MARINE_SMARTPREP)
 
 	listed_products = list(
-		/obj/item/storage/box/t26_system = list(CAT_ESS, "Essential Smartgunner Set", 0, "white"),
+		/obj/item/storage/box/t26_system = list(CAT_ESS, "T-26 Set", 0, "white"),
+		/obj/item/storage/box/t26_system/t25 = list(CAT_ESS, "T-25 Set", 0, "white"),
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:1085]: Can you add something that can make me differentiate these apart from a literal number

Comment (Pariah919) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:1085]: ok

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 2]
```diff
@@ -1081,7 +1081,9 @@ GLOBAL_LIST_INIT(marine_selector_cats, list(
 	req_access = list(ACCESS_MARINE_SMARTPREP)
 
 	listed_products = list(
-		/obj/item/storage/box/t26_system = list(CAT_ESS, "Essential Smartgunner Set", 0, "white"),
+		/obj/item/storage/box/t26_system = list(CAT_ESS, "T-26 Set", 0, "white"),
```
Comment (Pariah919) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:1084]:
```suggestion
		/obj/item/storage/box/t26_system = list(CAT_ESS, "T-26 Smartmachinegun Set", 0, "white"),
```

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 1 above)
Comment (Pariah919) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:1085]:
```suggestion
		/obj/item/storage/box/t26_system/t25 = list(CAT_ESS, "T-25 Smartrifle Set", 0, "white"),
```

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -971,7 +971,7 @@ datum/ammo/bullet/revolver/tp44
 	hud_state_empty = "smartgun_empty"
 	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	accurate_range = 15
-	damage = 20
+	damage = 25
```
Comment (SplinterGP) [inline code/modules/projectiles/ammo_datums.dm:974]: undocumented balance change

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_datums.dm (hunk 3 above)
Comment (Pariah919) [inline code/modules/projectiles/ammo_datums.dm:974]:
```suggestion
	damage = 20
```

**Comment by neohenry** [comment]
- Crate for T-25 spawns with 3 mags in total whereby T-29 spawns with 4 drums in total.
- Vendors that are open for everyone spawn in 2 T-29 drums
- Can do more damage

File: code/modules/projectiles/guns/rifles.dm [hunk 4]
```diff
@@ -911,6 +911,60 @@
 	starting_attachment_types = list(/obj/item/attachable/stock/t29stock, /obj/item/attachable/t29barrel, /obj/item/attachable/magnetic_harness)
 	gun_iff_signal = list(ACCESS_IFF_DS)
 
+
+// T-25 Smartrifle
+
+/obj/item/weapon/gun/rifle/standard_smartrifle
+	name = "\improper T-25 smartrifle"
+	desc = "The T-25 is the TGMC's current standard IFF-capable  rifle. It's known for its ability to lay down quick fire support very well. Requires special training and it cannot turn off IFF. It uses 10x26mm ammunition."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:919]:
```suggestion
	desc = "The T-25 is the TGMC's experimental rifle with IFF capabilities. It's known for its ability to lay down quick fire support very well. Requires special training and it cannot turn off IFF. It uses 10x26mm ammunition."
```

**Review by SplinterGP — APPROVED** [review]
seems all good to me

File: code/game/objects/items/storage/marine_boxes.dm [hunk 5]
```diff
@@ -19,6 +19,24 @@
...
+	slowdown = 1
+	can_hold = list() //Nada. Once you take the stuff out it doesn't fit back in.
+	foldable = null
+
+/obj/item/storage/box/t25_system/Initialize(mapload, ...)
+	. = ..()
+	new /obj/item/clothing/glasses/night/m56_goggles(src)
+	new /obj/item/weapon/gun/rifle/standard_smartrifle(src)
+	new /obj/item/storage/belt/marine/t25(src)
+	new /obj/item/ammo_magazine/rifle/standard_smartrifle(src)
```
Comment (SplinterGP) [inline code/game/objects/items/storage/marine_boxes.dm:38]:
```suggestion
	new /obj/item/storage/belt/marine/t25(src)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**